### ISSUE TITLE
Not working in subdir behind reverse proxy

### ISSUE THREAD

**Issue by Huljo** [RocketChat/Rocket.Chat#4546]
Rocket.Chat version: 0.42.0 / Manual CentOS install / NGINX

I can't run Rocket.chat in subdir. 
My ROOT_URL=https://domain.com/chat
When I attempt to load rocket chat from https://domain.com/chat i only get a 'Unknown path" response.

When i set My ROOT_URL to "https://domain.com/" I just get a loading status forever, assets are all absolute URLed to '/'.

My nginx config (rocket.chat is running at localhost:3000) : 

```
    location /chat/ {
       proxy_pass          http://127.0.0.1:3000/;
    }
```

Seems pretty similar to #2646 and #4460 but none of the proposed solutions are working for me.

**Comment by Huljo** [comment]
Tested on 0.43.0. Same problem.

**Comment by Huljo** [comment]
Please help me :(

**Comment by alvarow** [comment]
+1 ...

**Comment by Huljo** [comment]
One trick is to place this nginx conf:
```
location / {
    if ($arg_meteor_css_resource = "true") {
        rewrite ^ /search/$uri last;
    }
    if ($arg_meteor_js_resource = "true") {
        rewrite ^ /search/$uri last;
    }

    location /sockjs {
        rewrite ^ /search/$uri last;
    }
}
```

**Comment by Huljo** [comment]
Thank you to @abernix from Meteor project ([issue](https://github.com/meteor/meteor/issues/7916) for pointing out that if `settings.CDN_PREFIX` is defined, `setBundledJsCssPrefix` helper is used to rewrite urls and not `bundledJsCssUrlRewriteHook`, the default implementation. So subfolder can't be set before URLs during boilerplate template instantiation.

**Comment by MaluNoPeleke** [comment]
Has this problem already been solved?

**Comment by Huljo** [comment]
No, I actually set my subfolder as a CDN_PREFIX  to access to my rocket.chat server.

**Comment by MaluNoPeleke** [comment]
Can you describe how you did that?

**Comment by Huljo** [comment]
Open mongo shell on your server and type:
````
use {ROCKETCHAT_DATABASE_NAME}
db.rocketchat_settings.update({ "_id" : "CDN_PREFIX"}, {value: "/{SUBDIR_PATH}"})
````
This is not a fix, just a dirty way to access your app with the actual release when you work on a subfolder and voice calls/file uploads will still not working.

**Comment by alvarow** [comment]
Seems fixed on 0.46.0! Thanks
